### PR TITLE
github-action: export github actions telemetry traces

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Export GitHub actions execution as OpenTelemetry traces

## Why is it important?

This will help to gather insights and contribute to the GitHub action that empowers this functionality.

I also additionally, fixed some cosmetic changes in a couple of MacOS GitHub actions regarding their name and folder where to run the tests

## Issue

Similar to https://github.com/elastic/apm-agent-nodejs/pull/2805

## Tasks

- [x] Secrets have been created. 

## Follow up

To ingest the test-data in a follow up. Since all the GitHub actions together won't work

### CODE REVIEW

**Review by belimawr — APPROVED** [review]

**Review by amannocci — APPROVED** [review]
LGTM

File: .github/workflows/macos-xpack-osquerybeat.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: x-pack-heartbeat
+name: x-pack-osquerybeat
```
Comment (cachedout) [inline .github/workflows/macos-xpack-osquerybeat.yml:1]: I assume that these and similar changes are just to align the file name and the name in the YAML and that they are not related to the telemetry traces?

Comment (v1v) [reply .github/workflows/macos-xpack-osquerybeat.yml:1]: Exactly, I mixed up two different concerns in this PR, I'll add a note

**Review by cachedout — APPROVED** [review]

File: .github/workflows/macos-xpack-osquerybeat.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ on:
       - 8.*
 
 env:
-  BEAT_MODULE: 'x-pack/heartbeat'
+  BEAT_MODULE: 'x-pack/osquerybeat'
```
Comment (v1v) [inline .github/workflows/macos-xpack-osquerybeat.yml:14]: Changes are not related to the telemetry traces but a bug in the folder name

File: .github/workflows/macos-xpack-packetbeat.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-name: x-pack-metricbeat
+name: x-pack-packetbeat
```
Comment (v1v) [inline .github/workflows/macos-xpack-packetbeat.yml:1]: Changes are not related to the telemetry traces but a bug in the action name

File: .github/workflows/macos-xpack-packetbeat.yml [hunk 4]
```diff
@@ -11,7 +11,7 @@ on:
       - 8.*
 
 env:
-  BEAT_MODULE: 'x-pack/metricbeat'
+  BEAT_MODULE: 'x-pack/packetbeat'
```
Comment (v1v) [inline .github/workflows/macos-xpack-packetbeat.yml:14]: Changes are not related to the telemetry traces but a bug in the folder name

File: .github/workflows/macos-xpack-osquerybeat.yml (hunk 1 above)
Comment (v1v) [reply .github/workflows/macos-xpack-osquerybeat.yml:1]: Changes are not related to the telemetry traces but a bug in the folder name
